mne/decoding/tests/test_time_gen.py

```diff
@@ -205,33 +206,45 @@ def test_generalization_across_time():
 
     # Test testing time parameters
     # --- outside time range
-    gat.test_times = dict(start=-999.)
-    assert_raises(ValueError, gat.predict, epochs)
-    gat.test_times = dict(start=999.)
-    assert_raises(ValueError, gat.predict, epochs)
+    with warnings.catch_warnings(record=True) as w:
+        # This scenario use has len(y) < n_folds and thus throws a warning
```

Comment:
you need to put two `warnings.simplefilter`. One at the top of the file and one in each `warnings.catch_warnings`. Look at how we have done it elsewhere